index.android.js

```diff
@@ -1,98 +1,126 @@
-var React = require('react');
-var ReactNative = require('react-native');
+import React, {
+  PropTypes,
+  Component
+} from 'react';
 
-var {
-  NativeModules,
-  StyleSheet,
+import {
   View,
+  Text,
+  StyleSheet,
+  NativeModules,
   TouchableHighlight,
-  Text
-} = ReactNative;
+} from 'react-native';
 
-var FBLoginManager = NativeModules.FBLoginManager;
+import itypeof from 'itypeof';
 
-var itypeof = require('itypeof');
+const { FBLoginManager } = NativeModules;
 
-var FBLogin = React.createClass({
-  childContextTypes: {
-    isLoggedIn: React.PropTypes.bool,
-    login: React.PropTypes.func,
-    logout: React.PropTypes.func,
-    props: React.PropTypes.object
+const styles = StyleSheet.create({
+  login: {
+    flex: 1,
+    backgroundColor: '#3B5998',
+    padding: 10,
+    alignItems: 'center'
   },
-  getChildContext: function () {
+  whiteFont: {
+    color: 'white'
+  }
+});
+
+const statics = {
+  loginText: 'Login with Facebook',
+  logoutText: 'Logout from Facebook'
+};
+
+class FBLogin extends Component {
+  constructor (props) {
+    super(props);
+
+    this.login = this.login.bind(this);
+    this.logout = this.logout.bind(this);
+    this._handleEvent = this._handleEvent.bind(this);
+    this._getButtonView = this._getButtonView.bind(this);
+    this.getChildContext = this.getChildContext.bind(this);
+    this._onFacebookPress = this._onFacebookPress.bind(this);
+
+    this.state = {
+      statics:statics,
+      isLoggedIn: false,
+      buttonText: statics.loginText
+    };
+  }
+
+  componentDidMount(){
+    FBLoginManager.setLoginBehavior(self.props.loginBehavior);
+    FBLoginManager.getCredentials((data) => {
+      if(data &&
+        itypeof(data.credentials) === 'object' &&
+        itypeof(data.credentials.token) === 'string' &&
+        data.credentials.token.length > 0) {
+        this.setState({isLoggedIn:true, buttonText: self.state.statics.logoutText});
+      } else {
+        this.setState({isLoggedIn:false, buttonText: self.state.statics.loginText});
+      }
+      this._handleEvent(null,data);
+    })
+  }
+
+  static childContextTypes = {
+    isLoggedIn: PropTypes.bool,
+    login: PropTypes.func,
+    logout: PropTypes.func,
+    props: PropTypes.object
+  }
+
+  getChildContext () {
     return {
       isLoggedIn: this.state.isLoggedIn,
       login: this.login,
       logout: this.logout,
       props: this.props
     };
-  },
-  getInitialState() {
-   var statics = {
-     loginText: 'Login with Facebook',
-     logoutText: 'Logout from Facebook'
-   };
-  return {
-    statics:statics,
-    isLoggedIn: false,
-    buttonText: statics.loginText
-  };
-},
-logout() {
+  }
+
+  login(permissions) {
+    FBLoginManager.loginWithPermissions(
+      permissions || this.props.permissions,
+      (err,data) => this._handleEvent(err,data)
+    );
+  }
+
+  logout() {
```

Comment:
login and logout nee to be static method so they can be exposed for external use